docs/onboarding_best_practices.rst

```diff
@@ -0,0 +1,165 @@
+
+- For offline, perform the aggregations in your data preparation hadoop/spark job. 
+- For realtime, a samza job can be used to aggregate data at intervals based on the freshness requirements of the usecase.
+
+2. Time column
+###################
+Think about what **granularity of time column** is needed for your application. This will usually be *HOURS, DAYS, (sometimes 15 MINUTES)*. It is not recommended to have the time column in MILLISECONDS or SECONDS granularity. If time granularity is in milliseconds, the cardinality of the time column becomes very high in realtime systems, causing the time column dictionary to get very big. Bucket your time column to the coarsest granularity possible (hoursSinceEpoch, daysSinceEpoch). A greater level of pre-aggregations can be achieved with coarser time granularity.
+
+If you are unsure of the granularity, or feel that the chosen granularity might have to be made finer at a later point, you can keep the column in *MILLISECONDS*, however, round off the column to a higher granularity (round off millis to nearest day for example). This way you still have flexibility later on to change the bucketing.
+
+3. Segment Push Modes
+######################
+Pinot has 2 modes of segment push. *APPEND and REFRESH*.
+
+- APPEND - every data push is an incremental one, and the data received is appended to the existing data.  
+- REFRESH - The existing data is completely replaced by the new data pushed. The new segment names need to be exactly the same for the existing segments to be replaced
+
+In general, if you push a segment with the same name, it will overwrite the previous segment (in either mode).
+
+4. Prefer long over string where possible
+##########################################
+Strings in Java are inefficient. We end up serializing/deserializing them during query execution a lot of time.
+
+Instead of storing a long string value, store an id, and use a stateless utility function for conversion. Similarly, for attributes of a dimension (*for example, companyId -> company details*) store only the id in Pinot, and use another key value store/pinot table for the lookup. This is applicable only if Pinot cannot handle desired latency/throughput, even after all other possible optimizations. The reason is that, inside Pinot, we don’t know the relation between different columns. If used in a query like group by (group by companyId, companyName, companySize)  we will try to gather all the combinations of these columns in order to get the group-by results. This is quite expensive inside Pinot, but with key-value store service (or even keep a separate Pinot table), this will be simply a value lookup, which is way cheaper, and can definitely give you better performance
```

Comment:
"Instead of storing a long string, consider storing an id and using a stateless utility function for conversion if possible.

I think this paragraph needs re-work. I am not sure what is being said here.